### PR TITLE
Rename Log API to Log SPI to prevent confusion

### PR DESCRIPTION
**Pull request by tigrannajaryan**
We keep seeing confusion about what Log API is. There was a proposal [1]
to use the term SPI (Service Provider Interface) to help prevent the
confusion.

Please take a look and comment on whether you think this renaming helps
and is reasonable.

[1] https://github.com/open-telemetry/opentelemetry-specification/issues/2911#issuecomment-1421936787

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
This should be only merged if we see lots of approvals.

File: specification/logs/spi.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Logs API Interface
+# Logs SPI Interface
```
Comment (yurishkuro) [inline specification/logs/spi.md:1]:
```suggestion
# Logs SPI (Service Provider Interface)
```

**Review by yurishkuro — APPROVED** [review]

**Review by reyang — APPROVED** [review]
LGTM.

**Comment by MSNev** [comment]
I don't think service provider interface (SPI) is the correct "name" for this rename...

**Review by alanwest — APPROVED** [review]
This appears to mostly be a cosmetic change to help further draw people's attention to the fact that this is not an end-user facing API like the trace and metric API. In other words, it seems this new naming will have no affect on implementations.

I'm open to other names, but SPI seems ok.

"Log Appender API" seems too restrictive since the important aspect of this API is that it provides a shim between OpenTelemetry and some other software component - log appenders being perhaps only one type of component.

Log Shim API? ... SAPI 😆 kidding..

**Comment by reyang** [comment]
> "Log Appender API" seems too restrictive since the important aspect of this API is that it provides a shim between OpenTelemetry and some other software component - log appenders being perhaps only one type of component.
> 
> Log Shim API? ... SAPI 😆 kidding..

adapter/appender/shim/bridge/interop 🤷‍♂️

**Review by jsuereth — APPROVED** [review]

**Comment by jack-berg** [comment]
I'd prefer "Log Bridge API".

The goal of this PR to more clearly convey that this is not an API for end users. I don't think "SPI" accomplishes this. In fact, I'm not sure what I come away thinking when I read "Log SPI". From a java perspective, SPI is a sort of pluggable interface which can be provided by the app or third party to change the behavior of a library / framework. The appenders we've written sort of function like this, but don't actually use the Java SPI mechanism. If "SPI" is a java-centric term, but the java log appenders don't function as SPI implementations, then I'm not sure what we're conveying with this name.

Additionally, "API" is a [well established](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#api) term in OpenTelemetry. By introducing a new term like "SPI" we need to be able to define how SPIs are similar / different from an API. Do we expect additional SPIs in the future? Under what conditions do we call something an SPI?

In contrast, "Log Bridge API" keeps the existing "API" vocabulary, while clearly and succinctly conveying its purpose, as it's unlikely a user who comes across something called "bridge API" would think that they're supposed to use it.

I don't mind other terms like "adapter", "appender", "shim", but prefer "bridge" as its the most clear IMO and is quite succinct. Additionally, we use the term "bridge" similarly in [other places in the specification](https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-specification%20bridge&type=code).

**Comment by tigrannajaryan** [comment]
I like "Bridge API" more than "SPI" too. It reflects the intent more directly. Here is a PR with that version: https://github.com/open-telemetry/opentelemetry-specification/pull/3197

**Comment by tigrannajaryan** [comment]
Closing this in favour of https://github.com/open-telemetry/opentelemetry-specification/pull/3197 which received much wider support.

**Pull request closed without merging**